Apps/WebClient/src/NewClientApp/src/components/shared/MessageModalComponent.vue

```diff
@@ -0,0 +1,94 @@
+﻿<script setup lang="ts">
+import { ref } from "vue";
+import HgButtonComponent from "@/components/shared/HgButtonComponent.vue";
+import HgIconButtonComponent from "@/components/shared/HgIconButtonComponent.vue";
+
+interface Props {
+    title: string;
+    message: string;
+    isLoading?: boolean;
+    okOnly?: boolean;
+}
+withDefaults(defineProps<Props>(), {
+    isLoading: false,
+    okOnly: false,
+});
+
+const emit = defineEmits<{
+    (e: "submit"): void;
+    (e: "cancel"): void;
+}>();
+
+defineExpose({ showModal, hideModal });
+
+const isVisible = ref(false);
+
+function showModal(): void {
+    isVisible.value = true;
+}
+
+function hideModal(): void {
+    isVisible.value = false;
+}
+
+function handleSubmit(): void {
+    hideModal();
+    emit("submit");
+}
+
+function handleCancel(): void {
+    hideModal();
+    emit("cancel");
+}
+</script>
+
+<template>
+    <v-dialog
+        id="genericMessage"
+        v-model="isVisible"
+        persistent
+        no-click-animation
+    >
+        <div class="d-flex justify-center">
+            <v-card :loading="isLoading" max-width="700px">
+                <template v-slot:loader="{ isActive }">
+                    <v-progress-linear
+                        :active="isActive"
+                        color="primary"
```

Comment:
```suggestion
                        color="accent"
```
A primary-coloured loading indicator will not be visible against the background of the title bar (also primary colour).